api/src/wfl/service/firecloud.clj

```diff
@@ -126,6 +127,81 @@
                               {:Content/type "text/tab-separated-values"
                                :entities     (slurp file)})})))
 
+(defn import-entity-set
+  "
+  Formats existing sample ENTITIES into a Terra-compatible tsv
+  and uploads to WORKSPACE as an entity set entry identified by SET-NAME.
+
+  All entity references should correspond to existing entities.
+  All entity references should be of the same type.
```

Comment:
This was an assumption I made but wanted to verify.  If so, I can also enforce this via a precondition.

Reply:
I don't think you need to enforce this - firecloud will probably error if you try to do this!

Reply:
Just a suggestion …

```clojure
(defn import-entity-set
  "Format sample ENTITIES into a TSV, and upload it to WORKSPACE as an
  entity set named SET-NAME.  WORKSPACE is a \"namespace/name\" string.
  ENTITIES must share the same type.

  (import-entity-set \"workspace-namespace/workspace-name\"
                     \"c25ee04b-...\"
                     [[\"sample\" \"NA12878\"] [\"sample\" \"NA12879\"]])"
  [workspace set-name [entity & _more :as entities]]
  (let [[type _name] entity]
    (->> entities
         (map (partial vector set-name))
         (util/columns-rows->terra-tsv :membership [type type])
         .getBytes
         (import-entities workspace))))
```

Reply:
Should this generate `set-name`?

Reply:
Only 2 columns?  Both named `type`?
Ah. Maybe there is only 1 column.

Reply:
> Should this generate `set-name`?

No, we want access to this name further upstream so that we can specify it in submission creation, among other things (testing applications).

> Only 2 columns? Both named `type`?
> Ah. Maybe there is only 1 column.

I admittedly don't love this, but these base column names are then formatted by the `util/columns-rows->terra-tsv`.  For example `type=sample`:
- First column gets transformed into the Terra-compatible primary key column name, i.e. `membership:sample_set_id`
- Second column undergoes no further transformation on our end, but Terra needs it to match the passed-in entity type.  The resulting entity set table will see this column pluralized, i.e. `sample -> samples`.

Reply:
Thanks for the imperative mood.  Feel free to ignore the rest of my docstring comments as we move to a new standard.